python/graphstorm/model/gnn.py

```diff
@@ -88,17 +88,34 @@ def __init__(self, dense_opts=None, lm_opts=None, sparse_opts=None):
         for optimizer in all_opts:
             assert optimizer is not None
 
-    def zero_grad(self):
+    def _clear_traces(self):
+        """ Clear the traces in sparse optimizers.
+        """
+        for optimizer in self.sparse_opts:
+            for emb in optimizer._params:
+                emb.reset_trace()
+
+    def zero_grad(self, optimize_sparse_params=True):
```

Comment:
there is a problem here. if we add this argument, the interface of the optimizer is not the same as the regular optimizer. can we consider the implementation specific to GLEM? should we implement a GLEM optimizer?

Reply:
Added `GLEMOptimizer`